src/main/server/player/server-player-component.ts

```diff
@@ -137,13 +142,19 @@ export class ServerPlayerComponent {
 
    private handlePlayerPositionChanged(playerId: string, position: Vector): void {
       this.collisionPhysics.updatePosition(playerId, position.x, position.y);
-      // TODO: Check lava distance and kill if closer
       this.ammo.pickupInRectangle(
          playerId,
          position.x,
          position.y,
          ServerConfig.PLAYER_WIDTH,
          ServerConfig.PLAYER_HEIGHT,
       );
+      if (this.lavaDistance2(position) <= ServerConfig.LAVA_RADIUS * ServerConfig.LAVA_RADIUS * 0.9) {
```

Comment:
should this not check if the player's `x,y` is less than the circumference of the circle?

https://www.w3resource.com/javascript-exercises/javascript-basic-exercise-120.php

Reply:
It does exactly this one doesn't it? Apartm from the typo. X2+y2<r2 cos the other point is 0,0